.github/workflows/ci.yaml

```diff
@@ -15,26 +15,11 @@ jobs:
         go-version: ~1.17
       id: go
 
-    - name: Cache the go module cache
-      id: cache-go-mod
-      uses: actions/cache@v2
-      with:
-        path: ~/go/pkg/mod
-        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-        restore-keys: |
-          ${{ runner.os }}-go-
-
-    - name: Get dependencies
-      run: go mod download
-
     - name: Run verification checks
       run: make verify
 
-    - name: Run unit tests
-      run: make test
-
-    - name: Verify controller binary can compile
-      run: make build
+    - name: Verify controller container image can build
+      run: make docker-build
```

Comment:
This commands runs the `test` Makefile target as a prerequisite, so we should be fine to drop the "Run unit tests" job, and the "Verify controller binary can compile" job as we're implicitly verifying that when building the binary in the container image environment.